lib/core/importer.ts

```diff
@@ -73,4 +73,19 @@ export class Importer {
     );
     return globs.flat();
   }
+
+  /**
+   * This code is intentionally structured to execute the callback function registered with `.then()`
+   * and proceed through the event loop only after the asynchronous task of the `locate` function is completed.
+   * Designed to handle potential errors after the scan is completed.
+   */
+  private catchOverlappedScanScope(value: { new (...args: any[]): any; name: string }, pathName: string) {
+    locate(value as any).then(({ path }: { path: string }) => {
+      if (!this.rootPath) this.rootPath = pathName;
+      if (this.rootPath !== path) {
+        new Logger(value.name).error('[@ComponentScan] module scope cannot be overlapped', `${this.rootPath}\n${path}`);
```

Comment:
if has multiple overlapped value, it'll catch just foremost one.
I think use internal queue for catch multiple overlapped paths, and notify to developer with scanning all project.

Reply:
Good point. but this will be totally taken care of in a different PR!